railties/lib/rails/backtrace_cleaner.rb

```diff
@@ -4,18 +4,26 @@
 
 module Rails
   class BacktraceCleaner < ActiveSupport::BacktraceCleaner
-    APP_DIRS_PATTERN = /^\/?(app|config|lib|test|\(\w*\))/
+    APP_DIRS_PATTERN = /\A\/?(?:app|config|lib|test|\(\w*\))/
     RENDER_TEMPLATE_PATTERN = /:in `.*_\w+_{2,3}\d+_\d+'/
-    EMPTY_STRING = ""
-    SLASH        = "/"
-    DOT_SLASH    = "./"
 
     def initialize
       super
       @root = "#{Rails.root}/"
-      add_filter { |line| line.sub(@root, EMPTY_STRING) }
-      add_filter { |line| line.sub(RENDER_TEMPLATE_PATTERN, EMPTY_STRING) }
-      add_filter { |line| line.sub(DOT_SLASH, SLASH) } # for tests
+      add_filter do |line|
+        line.start_with?(@root) ? line[@root.size, line.size] : line
+      end
+      add_filter do |line|
+        if RENDER_TEMPLATE_PATTERN.match?(line)
+          line.sub(RENDER_TEMPLATE_PATTERN, "")
+        else
+          line
+        end
+      end
+      add_filter do |line|
+        # line.sub(/\A\.\//, "/")
+        line.start_with?("./") ? line[1, line.size] : line
```

Comment:
Noticed that the previous comment said `# for tests`, since GitHub's backtraces are so large would there be any gain in creating a custom `backtrace_cleaner` for our tests? Caveat is that I don't know quite when `./` is prefixed.

Reply:
I guess the `sub` comment can go too.

Reply:
I'm actually just testing/benchmarking against vanilla Rails's backtraces 😆.

Looking at this filter again, I don't think we actually need/want it. I'll clean this up before merging and open another PR to discuss removing it (also requires a fix to the silencer).